lua/src/macro_cache.cc

```diff
@@ -182,37 +251,94 @@ void macro_cache::_process_instance(instance_broadcast const& in) {
  */
 void macro_cache::_process_host(neb::host const& h) {
   logging::debug(logging::medium)
-    << "macro_cache: process host --- "
-    << "id = " << h.host_id
-    << " ; name = " << h.host_name;
+    << "lua: processing host '" << h.host_name << "' of id " << h.host_id;
   _hosts[h.host_id] = h;
 }
 
+/**
+ *  Process a host group event.
+ *
+ *  @param hg  The event.
+ */
+void macro_cache::_process_host_group(neb::host_group const& hg) {
+  logging::debug(logging::medium)
+    << "lua: processing host group '" << hg.name << "' of id " << hg.id;
+  if (hg.enabled)
+    _host_groups[hg.id] = hg;
+  else
+    _host_groups.remove(hg.id);
+}
+
+/**
+ *  Process a host group member event.
+ *
+ *  @param hgm  The event.
+ */
+void macro_cache::_process_host_group_member(
+       neb::host_group_member const& hgm) {
+  logging::debug(logging::medium)
+    << "lua: processing host group member "
+    << " (group_name: '" << hgm.group_name << "', group_id: " << hgm.group_id
+    << ", host_id: " << hgm.host_id << ")";
+  if (hgm.enabled)
+    _host_group_members.insert(hgm.host_id, hgm);
+  else
+    _host_group_members.remove(hgm.host_id);
+}
+
 /**
  *  Process a service event.
  *
  *  @param s  The event.
  */
 void macro_cache::_process_service(neb::service const& s) {
   logging::debug(logging::medium)
-    << "macro_cache: process service --- "
-    << "host id = " << s.host_id
-    << " ; service id = " << s.service_id
-    << " ; description = " << s.service_description;
+    << "lua: processing service (" << s.host_id << ", " << s.service_id << ") "
+    << "(description: " << s.service_description << ")";
   _services[qMakePair(s.host_id, s.service_id)] = s;
 }
 
+/**
+ *  Process a service group event.
+ *
+ *  @param sg  The event.
+ */
+void macro_cache::_process_service_group(neb::service_group const& sg) {
+  logging::debug(logging::medium)
+    << "lua: processing service group '" << sg.name << "' of id " << sg.id;
+  if (sg.enabled)
+    _service_groups[sg.id] = sg;
+  else
+    _service_groups.remove(sg.id);
+}
+
+/**
+ *  Process a service group member event.
+ *
+ *  @param sgm  The event.
+ */
+void macro_cache::_process_service_group_member(
+       neb::service_group_member const& sgm) {
+  logging::debug(logging::medium)
+    << "lua: processing service group member "
+    << " (group_name: '" << sgm.group_name << "', group_id: " << sgm.group_id
+    << ", host_id: " << sgm.host_id
+    << ", service_id: " << sgm.service_id << ")";
+  if (sgm.enabled)
+    _service_group_members.insert(qMakePair(sgm.host_id, sgm.service_id), sgm);
```

Comment:
My comments about host group membership insertion / removal also apply to service group memberships.